kale/prepdata/supergraph_construct.py

```diff
@@ -0,0 +1,250 @@
+        edge_type (torch.Tensor, optional): one-dimensional relation type for each edge, indexed from 0. Defaults to None.
+        edge_weight (torch.Tensor, optional): one-dimensional weight for each edge. Defaults to None.
+
+    Examples:
+        >>> import torch
+        >>> node_feat = torch.randn(4, 20)
+        >>> edge_index = torch.tensor([[0, 1, 2, 3], [1, 2, 3, 0]])
+        >>> edge_type = torch.tensor([0, 0, 1, 1])
+        >>> edge_weight = torch.randn(4)
+        >>> # create a supervertex with homogeneous edges
+        >>> supervertex_homo = SuperVertex(node_feat, edge_index)
+        >>> # create a supervertex with hetegeneous edges
+        >>> supervertex_hete = SuperVertex(node_feat, edge_index, edge_type)
+        >>> # create a supervertex with weighted edges
+        >>> supervertex_weight1 = SuperVertex(node_feat, edge_index, edge_weight=edge_weight)
+        >>> supervertex_weight2 = SuperVertex(node_feat, edge_index, edge_type, edge_weight)
+    """
+
+    def __init__(
+        self,
+        name: str,
+        node_feat: torch.Tensor,
+        edge_index: torch.Tensor,
+        edge_type: torch.Tensor = None,
+        edge_weight: torch.Tensor = None,
+    ) -> None:
+
+        self.name = name
+        self.node_feat = node_feat
+        self.edge_index = edge_index
+        self.edge_type = edge_type
+        self.edge_weight = edge_weight
+
+        # get the number of nodes, node features and edges
+        self.num_node, self.num_node_feat = node_feat.shape
+        self.num_edge = edge_index.shape[1]
+
+        # initialize in-supervertex and out-supervertex lists
+        self.in_supervertex_list: List[str] = []
+        self.out_supervertex_list: List[str] = []
+        self.if_start_supervertex = True
+
+        self.__process_edges__()
+
+    def __process_edges__(self):
+        r"""
+        process the edges of the supervertex.
+        """
+        # get the number of edge types
+        if self.edge_type is None:
+            self.num_edge_type = 1
+        else:
+            unique_edge_type = self.edge_type.unique()
+            self.num_edge_type = unique_edge_type.shape[0]
+
+            # check if the index of edge type is continuous and starts from 0
+            if self.num_edge_type != unique_edge_type.max() + 1:
+                error_msg = "The index of edge type is not continuous and starts from 0."
+                logging.error(error_msg)
+                raise ValueError(error_msg)
+
+            # sort the edges and edge types
+            sort_index = torch.argsort(self.edge_type)
+            self.edge_index = self.edge_index[:, sort_index]
+            self.edge_type = self.edge_type[sort_index]
+
+            self.__get_range_list__()
+
+    def __get_range_list__(self):
+        """get the range of edge types"""
+        idx = 0
+        range_list = [[0, 0]]
+
+        for i, et in enumerate(self.edge_type):
```

Comment:
`et` is too brief. Make it more readable, e.g., `edgetype`?

Reply:
Changed `et` to `edgetype` in [5c193a3](https://github.com/pykale/pykale/pull/338/commits/5c193a39c6e3d76aa9249ae5ff9546a17d6c60a2).